packages/terra-layout/src/Layout.jsx

```diff
@@ -0,0 +1,193 @@
+    this.renderMenu = this.renderMenu.bind(this);
+    this.renderContent = this.renderContent.bind(this);
+
+    this.state = Layout.stateForProps(props, {
+      size: getBreakpointSize(),
+    });
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this.updateSize);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState(Layout.stateForProps(nextProps, this.state));
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.updateSize);
+  }
+
+  updateSize() {
+    const newSize = getBreakpointSize();
+
+    if (this.state.size !== newSize) {
+      this.setState(Layout.stateForProps(this.props, {
+        size: newSize,
+      }));
+    }
+  }
+
+  toggleMenu() {
+    this.setState({
+      menuIsOpen: !this.state.menuIsOpen,
+    });
+  }
+
+  togglePin() {
+    this.setState({
+      menuIsPinned: !this.state.menuIsPinned,
+    });
+  }
+
+  renderHeader() {
+    const { header } = this.props;
+    const { size, menuIsOpen, isToggleMenu, menuIsPresent } = this.state;
+
+    if (!header) {
+      return null;
+    }
+
+    const shouldAllowMenuToggle = isToggleMenu && menuIsPresent;
+
+    return React.cloneElement(header, {
+      layoutConfig: {
+        size,
+        toggleMenu: shouldAllowMenuToggle ? this.toggleMenu : undefined,
+        menuIsOpen,
+      },
+    });
+  }
+
+  renderMenu() {
+    const { menu } = this.props;
+    const { size, menuIsOpen, menuIsPinned, isToggleMenu, isHoverMenu, menuIsPresent } = this.state;
+    const shouldAllowMenuToggle = isToggleMenu && menuIsPresent;
+
+    if (!menuIsPresent) {
+      return null;
+    }
+
+    return React.cloneElement(menu, {
+      layoutConfig: {
+        size,
+        toggleMenu: shouldAllowMenuToggle ? this.toggleMenu : undefined,
+        menuIsOpen,
+        togglePin: isHoverMenu ? this.togglePin : undefined,
+        menuIsPinned,
+      },
+    });
+  }
+
+  renderContent() {
+    const { children } = this.props;
+    const { size, menuIsOpen, isToggleMenu, menuIsPresent } = this.state;
+    const shouldAllowMenuToggle = isToggleMenu && menuIsPresent;
+
+    if (!children) {
+      return null;
+    }
+
+    return (
+      <ContentContainer
+        fill
+        header={isToggleMenu && this.renderHeader()}
+        style={{ outline: 'none' }}
+      >
+        {(
+          React.cloneElement(children, {
+            layoutConfig: {
+              size,
+              toggleMenu: shouldAllowMenuToggle ? this.toggleMenu : undefined,
+              menuIsOpen,
+            },
+          })
+        )}
+      </ContentContainer>
+    );
+  }
+
+  render() {
+    const { menuText } = this.props;
+    const { menuIsOpen, menuIsPinned, size, isHoverMenu, isFixedMenu, isToggleMenu, menuIsPresent } = this.state;
+
+    return (
+      <ContentContainer
+        fill
+        header={!isToggleMenu && this.renderHeader()}
+        {...getCustomProps(this.props, propTypes)}
```

Comment:
It seems this function is only used once in these changes. Would it just be easier to do
`const { menuText, header, menu, children, ...customProps } = props` instead? Do we expect to reuse this in future work?

Reply:
I added this function back when there were quite a few more props and it made more sense. I'd still like to keep it around because 1. I find it easier than having to cross-reference the propTypes to ensure we're listing everything and 2. I find working around the unused props lint warning annoying. I'd like to move the function into another package (like the previously discussed `terra-utitlities` or something ) but I have not yet been able to do so.